test/python/circuit/test_circuit_load_from_qpy.py

```diff
@@ -1456,15 +1457,16 @@ def test_incomplete_owned_bits(self):
 
     def test_diagonal_gate(self):
         """Test that a `DiagonalGate` successfully roundtrips."""
+        diag = DiagonalGate([1, -1, -1, 1])
+
         qc = QuantumCircuit(2)
-        with self.assertWarns(PendingDeprecationWarning):
-            qc.diagonal([1, -1, -1, 1], [0, 1])
+        qc.append(diag, [0, 1])
 
         with io.BytesIO() as fptr:
             dump(qc, fptr)
             fptr.seek(0)
             new_circuit = load(fptr)[0]
-        # DiagonalGate (and a bunch of the qiskit.extensions gates) have non-deterministic
+        # DiagonalGate (and some of the qiskit.circuit.library gates) have non-deterministic
```

Comment:
lol, I'm betting I wrote that "a bunch"

Reply:
haha I proposed to tone it down since now it's only a few gates from the standard library (whereas before it was lot of the ones in extension) 😛

Reply:
Haha yeah, that's fine.  I'm mostly laughing at my terminology - I think I'm the only one of us who talks like that, and almost certainly the only one who writes code like that.